plugins/modules/clusters.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+
+import os
+import requests
+
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: clusters
+
+short_description: Handle AssistedInstall clusters
+
+version_added: "1.0.0"
+
+description: Interact with clusters through the AssistedInstall API
+
+options:
+    action:
+      description: The action to perform (list, create, delete)
+      required: false
+      default: "list"
+    cluster_id:
+        description: The cluster ID to perform the action on
+        required: false
+    with_hosts:
```

Comment:
Should this be a sub-option as well under query_params?

Reply:
What is your idea for calling the module?

Currently it will work as follow with these inputs:

```yaml
- name: List clusters
  clusters:
    action: list
    with_hosts: true    # This one is optional, but when defined it is used in "list"
    cluster_id: "{{ some_cluster_id }}" # This is ignored in the "list" action.
```

```yaml
- name: Delete clusters
  clusters:
    action: delete
    cluster_id: "{{ cluster_id_to_delete }}"  # This option is required when the action is "delete"
    with_hosts: true    # This one *could* be defined but is ignored
```

What are your thoughts using a suboption in here? Can you share how it would look like, as the examples above?

Reply:
```suggestion
  -  name: List clusters
     clusters:
       query_params:
         with_hosts: true
         owner: val
         get_unregistered_clusters: true
    
```

Reply:
Above is what I was initially thinking, but looks like we can work without suboptions for the time being

Reply:
What about:

```yaml
  -  name: List clusters
     clusters:
      action: list
      list_params:
         with_hosts: true
         owner: val
         get_unregistered_clusters: true

```